staff/tests/views/coupon/test_coupon_new_batch.py

```diff
@@ -76,8 +110,21 @@ def test_coupon_generation_incorrect_headers_rerenders_form_page_with_appropriat
     uploaded_file = SimpleUploadedFile(name=csv_file.name, content=bytes(content, 'utf-8'), content_type="multipart/form-data")
 
     response = client.post(reverse('coupon_new_batch'), {'csv_file': uploaded_file})
-    print(response.context['errors'])
 
     assert response.status_code == HttpResponse.status_code
     assert 'coupon/new_batch.html' in (template.name for template in response.templates)
     assert 'The file you uploaded requires the specific headers "first_name" and "email"!' in response.context['errors'][0]
+
+def test_coupon_generation_creates_correct_coupons(logged_in_existing_user, coupon_effect_basics, coupon_effect_bootcamp):
+    test_file_path = "./staff/tests/forms/csv_files/correct_test_file.csv"
+    csv_file = open(test_file_path, 'r')
+    content = csv_file.read()
+    uploaded_file = SimpleUploadedFile(name=csv_file.name, content=bytes(content, 'utf-8'), content_type="multipart/form-data")
+
+    client.post(reverse('coupon_new_batch'), {'csv_file': uploaded_file})
+    coupons = Coupon.objects.all()
+
+    assert len(coupons) == 2
+    assert coupons.first().effects.all()[0] == coupon_effect_basics
+    assert coupons.first().effects.all()[1] == coupon_effect_bootcamp
```

Comment:
can also do the below. might be easier to read?
assert coupons.first().effects == [coupon_effect_basics, coupon_effect_bootcamp]

Reply:
the problem i'm facing is that `coupons.first().effects` refers to the many-to-many manager and `coupons.first().effects.all()` is a queryset and not a list